src/app/models/workspace.server.model.ts

```diff
@@ -0,0 +1,55 @@
+import { Mongoose, Schema } from 'mongoose'
+
+import { IWorkspaceModel, IWorkspaceSchema } from '../../types'
+
+export const WORKSPACE_SCHEMA_ID = 'Workspace'
+
+const compileWorkspaceModel = (db: Mongoose): IWorkspaceModel => {
+  const WorkspaceSchema = new Schema<IWorkspaceSchema, IWorkspaceModel>({
+    title: {
+      type: String,
+      validate: [
+        /^[a-zA-Z0-9_\-./() &`;'"]*$/,
+        'Workspace title cannot contain special characters',
+      ],
+      required: 'Workspace title cannot be blank',
+      minlength: [4, 'Workspace title must be at least 4 characters'],
+      maxlength: [200, 'Workspace title can have a maximum of 200 characters'],
+      trim: true,
+    },
+    admin: {
+      type: Schema.Types.ObjectId,
+      ref: 'User',
+      required: 'Workspace must have an Admin',
+    },
+    formIds: {
+      type: [Schema.Types.ObjectId],
+      validate: {
+        async validator(this: IWorkspaceSchema) {
+          const areFormIdsUnique =
+            new Set(this.formIds).size === this.formIds.length
+
+          return areFormIdsUnique
+        },
+      },
+      message: "Failed to update workspace document's formIds",
+    },
+  })
+
+  WorkspaceSchema.index({
+    admin: 1,
+  })
```

Comment:
might also want to add that in your PR so we remember to also add an index in our DB provider